main.tf

```diff
@@ -58,7 +58,7 @@ module "vnet" {
 
   name                = var.vnet_name
   prefix              = var.prefix
-  resource_group_name = module.resource_group.name
+  resource_group_name = var.vnet_resource_group_name == null ? module.resource_group.name : var.vnet_resource_group_name
```

Comment:
Though you call it `vnet_resource_group_name` it could be any resource group. So the naming should change.